### ISSUE TITLE
Gradient clipping

### ISSUE THREAD

**Issue by mikecmpbll** [mikecmpbll/rann#8]
Prevent exploding gradients with gradient clipping .. Sample gradients at beginning of training (perhaps?) and get an average of the expected gradient threshold. If subsequent gradients are above this threshold then scale them down:

```ruby
if grad.abs > threshold
   grad *= threshold/grad.abs
end
```
